cocos/renderer/CCRenderer.h

```diff
@@ -258,8 +258,12 @@ class CC_DLL Renderer
     // the TriBatches
     TriBatchToDraw* _triBatchesToDraw;
 
+    // used as offset in in the process of concatenating triangles for batched drawing
     int _filledVertex;
     int _filledIndex;
+    // used for the flushing logic if vertex number is high and flushing is needed
+    int _filledVertex2;
+    int _filledIndex2;
```

Comment:
According to [General Naming Rules within the Engine](https://github.com/cocos2d/cocos2d-x/wiki/Cpp--Coding-Style#general-naming-rules)

> Function names, variable names, and filenames should be descriptive; eschew abbreviation.
> Give as descriptive a name as possible, within reason. Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader. Do not use abbreviations that are ambiguous or unfamiliar to readers outside your project, and do not abbreviate by deleting letters within a word.

`_filledVertex2` and `_filledIndex2` implies that there are / there will be other variations (ex: `_filledVertex3`, `_filledVertex4` ... etc), that these are either elements of an array or a group

Reply:
@gestern Yes, will work on renaming variables after. For now I am only interested in you opinion on the overall solution. This solution does fix the crashes but I'm not sure if it can introduce other issues that I missed.